driver/src/main/com/mongodb/client/gridfs/GridFSBucket.java

```diff
@@ -0,0 +1,253 @@
+
+    /**
+     * Downloads the contents of the stored file specified by {@code id} and writes the contents to the {@code destination} Stream.
+     *
+     * @param id the custom id of the file, to be written to the destination stream
+     * @param destination the destination stream
+     * @deprecated using custom id values for with GridFS is no longer supported
+     */
+    @Deprecated
+    void downloadToStream(BsonValue id, OutputStream destination);
+
+    /**
+     * Opens a Stream from which the application can read the contents of the latest version of the stored file specified by the
+     * {@code filename}.
+     *
+     * @param filename the name of the file to be downloaded
+     * @return the stream
+     */
+    GridFSDownloadStream openDownloadStreamByName(String filename);
+
+    /**
+     * Opens a Stream from which the application can read the contents of the stored file specified by {@code filename} and the revision
+     * in {@code options}.
+     *
+     * @param filename the name of the file to be downloaded
+     * @param options the download options
+     * @return the stream
+     */
+    GridFSDownloadStream openDownloadStreamByName(String filename, GridFSDownloadByNameOptions options);
+
+    /**
+     * Downloads the contents of the latest version of the stored file specified by {@code filename} and writes the contents to
+     * the {@code destination} Stream.
+     *
+     * @param filename the name of the file to be downloaded
+     * @param destination the destination stream
+     */
+    void downloadToStreamByName(String filename, OutputStream destination);
+
+    /**
+     * Downloads the contents of the stored file specified by {@code filename} and by the revision in {@code options} and writes the
+     * contents to the {@code destination} Stream.
+     *
+     * @param filename the name of the file to be downloaded
+     * @param destination the destination stream
+     * @param options the download options
+     */
+    void downloadToStreamByName(String filename, OutputStream destination, GridFSDownloadByNameOptions options);
+
+    /**
+     * Finds all documents in the files collection.
+     *
+     * @return the GridFS find iterable interface
+     * @mongodb.driver.manual tutorial/query-documents/ Find
+     */
+    GridFSFindIterable find();
+
+    /**
+     * Finds all documents in the collection that match the filter.
+     *
+     * <p>
+     *     Below is an example of filtering against the filename and some nested metadata that can also be stored along with the file data:
+     *  <pre>
+     *  {@code
+     *      Filters.and(Filters.eq("filename", "mongodb.png"), Filters.eq("metadata.contentType", "image/png"));
+     *  }
+     *  </pre>
+     * </p>
+     *
+     * @param filter the query filter
+     * @return the GridFS find iterable interface
+     * @see com.mongodb.client.model.Filters
+     */
+    GridFSFindIterable find(Bson filter);
```

Comment:
MongoIterables are fluent so the GridFSFindOptions are apart of the GridFSFindIterable api.